### ISSUE TITLE
about venus dataset and venus_python dataset

### ISSUE THREAD

**Issue by likecake** [Elfsong/Afterburner#3]
Hi,

I noticed that in the Venus dataset, the Python subset contains the field "integral" for each solution.

However, in other language subsets (such as C++), there is no "integral" field. Only runtime and memory information seem to be available (e.g., in rt_list and mm_list).

How is the "integral" metric supposed to be calculated for these languages?

Is there a recommended formula to combine runtime and memory into an integral score?

Thanks!

**Comment by likecake** [comment]
and In the DPO dataset construction, why  use `selected_solutions = random.sample(solutions, 2)` instead of sorting all solutions by runtime and selecting the fastest and slowest ones?

**Comment by Elfsong** [comment]
1) Maybe you can read the paper at first. 'Integral' serves as a comprehensive metric that evaluates both runtime and memory efficiency, which is the **integral** of runtime and memory literally.

2) Pairing the absolute fastest and slowest solutions is suboptimal for DPO training, as these extremes tend to introduce highly unconventional coding styles into the training data.

**Comment by likecake** [comment]
I’m truly honored to receive responses from you.
From my shallow understanding,the integral score requires running the program and sampling runtime and memory usage during execution, which implies that the evaluation script maybe need execute the solution with a set of test cases.
However, in the Venus dataset, it looks like that the test cases are not included.
In this case, is it correct to assume that the integral metric cannot be computed directly for Venus unless the corresponding test cases are reconstructed?

**Comment by Elfsong** [comment]
Hi, you can check out [Venus_General_Test](https://huggingface.co/datasets/Elfsong/Venus_General_Test) for language-agnostic test cases.

**Comment by likecake** [comment]
OK,Thank you so much!

2026-03-10 18:27:39 "Mingzhe" ***@***.***> 写道：
Elfsong left a comment (Elfsong/Afterburner#3)

Hi, you can check out Venus_General_Test for language-agnostic test cases.
